src/NServiceBus.Core/Routing/FileBasedDynamicRouting/FileRoutingTable.cs

```diff
@@ -24,21 +22,10 @@ public FileRoutingTable(string filePath, TimeSpan checkInterval, IAsyncTimer tim
             errorMessage = $"An error occured while reading the endpoint instance mapping file at {filePath}. See the inner exception for more details.";
         }
 
-        protected override async Task OnStart(IMessageSession session)
-        {
-            var endpointInstances = settings.Get<EndpointInstances>();
-            endpointInstances.AddDynamic(FindInstances);
-
-            try
-            {
-                await ReloadData().ConfigureAwait(false);
-            }
-            catch (Exception ex)
-            {
-                throw new Exception(errorMessage, ex);
-            }
-
+        protected override Task OnStart(IMessageSession session)
+        {            
             timer.Start(ReloadData, checkInterval, ex => log.Error(errorMessage, ex));
```

Comment:
Would it make sense to move the timer start to the first load in `FindInstances` as well? That way the endpoint would only tries to load the file if the endpoint is actually trying to send messages.